### PR TITLE
Add nvm mod for code-server

### PR DESCRIPTION
**Pull request by fabricionaweb**
Since the last MR #311 was abandoned, I'm giving my try here.

### CODE REVIEW

File: root/etc/cont-init.d/98-nvm [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+echo "**** installing nvm ****"
+export NVM_DIR="$HOME/.nvm"
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+
+echo "**** loading nvm ****"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+
+echo "**** installing the latest release ****"
+nvm install node
```
Comment (fabricionaweb) [inline root/etc/cont-init.d/98-nvm:11]: Like the [documentation said](https://github.com/nvm-sh/nvm#usage) `The first version installed becomes the default. New shells will start with the default version of node`

File: root/etc/cont-init.d/98-nvm [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/with-contenv bash
+
+echo "**** installing nvm ****"
+export NVM_DIR="$HOME/.nvm"
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+
+echo "**** loading nvm ****"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
```
Comment (fabricionaweb) [inline root/etc/cont-init.d/98-nvm:8]: The install will automatically adds the load to `.bashrc` and `.zshrc`. 

Every new instance of `bash` or `zsh` should start with nvm (if everything went well...)

We are only loading the nvm at this point to be able to run the script bellow and install the first node version, which I think is cool.

But is also an option remove it (after line 6) and then start without any node version

File: root/etc/cont-init.d/98-nvm (hunk 1 above)
Comment (aptalca) [reply root/etc/cont-init.d/98-nvm:11]: this is fine

File: root/etc/cont-init.d/98-nvm (hunk 2 above)
Comment (aptalca) [reply root/etc/cont-init.d/98-nvm:8]: can you change this into a proper if block?

Comment (fabricionaweb) [reply root/etc/cont-init.d/98-nvm:8]: Yes I can, like this?

```suggestion
if [ -s "$NVM_DIR/nvm.sh" ]; then
  \. "$NVM_DIR/nvm.sh"
fi
```

To be honest I dont know if is necessary the if, like we are installing right above
Can we just load it?

```suggestion
\. "$NVM_DIR/nvm.sh"
```

Comment (aptalca) [reply root/etc/cont-init.d/98-nvm:8]: I guess. One question, does the install script need to run on every start? Is that how it is supposed to be updated?

Comment (fabricionaweb) [reply root/etc/cont-init.d/98-nvm:8]: No, does not need to run every start. Just once.

What is needed to load every start is the `nvm.sh` but the install script adds it to .bashrc and .zshrc

Comment (aptalca) [reply root/etc/cont-init.d/98-nvm]: Then you should put that install operation inside the if block so it only installs it the first time. NVM folder is in the config folder and should be persistent

Comment (fabricionaweb) [reply root/etc/cont-init.d/98-nvm]: Changed

**Comment by aptalca** [comment]
Did you test this? Because it doesn't work in its current form.

The NVM_DIR variable should be exported
Also, the install script adds the loading bits to `.profile` even though `.bashrc` is present. The issue is, for some reason, nvm doesn't get loaded from `.profile`, but if manually added to `.bashrc` it does get loaded.

I think you need to set this up manually instead of relying on their install script. And test it with both code-server and openvscode-server

**Comment by fabricionaweb** [comment]
Ok Im gonna do a proper test and I back later. 
Thanks you!

**Comment by fabricionaweb** [comment]
Ok I've run tests here. And seems working now 🎉 

There was a typo 👀 It was `if` when should be `fi` (at final) 

How I did test:
- Created and publish [the package on GitHub registry](https://github.com/fabricionaweb/docker-mods/pkgs/container/docker-mods)
- Create the container
  ```
  docker create \
  --name=code-server \
  -e DOCKER_MODS=ghcr.io/fabricionaweb/docker-mods \
  -e PUID=1000 \
  -e PGID=1000 \
  -e TZ=Europe/London \
  -p 8443:8443 \
  lscr.io/linuxserver/code-server
  ```
- Start the container for the first time
  ```
  docker start -a code-server
  ```
  <details>
  <summary>Analyze the logs (installing)</summary>

  ```
  [cont-init.d] 98-nvm: executing... 
  **** installing nvm ****
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload   Upload  Total   Spent    Left  Speed
  100 15687  100 15687    0     0  75782      0 --:--:-- --:--:-- --:--:-- 75782
  => Downloading nvm from git to '/config/.nvm'
  => Cloning into '/config/.nvm'...
  * (HEAD detached at FETCH_HEAD)
    master
  => Compressing and cleaning up git repository
  
  => Appending nvm source string to /config/.bashrc
  => Appending bash_completion source string to /config/.bashrc
  => Close and reopen your terminal to start using nvm or run the following to use it now:
  
  export NVM_DIR="$HOME/.nvm"
  [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
  [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
  **** loading nvm ****
  **** installing the latest release ****
  Downloading and installing node v18.2.0...
  Downloading https://nodejs.org/dist/v18.2.0/node-v18.2.0-linux-x64.tar.gz...
  ######################################################################## 100.0%
  100.0%
  Computing checksum with sha256sum
  Checksums matched!
  Now using node v18.2.0 (npm v8.9.0)
  Creating default alias: default -> node (-> v18.2.0)
  [cont-init.d] 98-nvm: exited 0.
  ```

  </details>

- Stop the container
- Start the container again
  <details>
  <summary>Analyze the logs again (did not install)</summary>

  ```
  [cont-init.d] 98-nvm: executing... 
  **** installing nvm ****
  **** nvm already installed, skipping ****
  [cont-init.d] 98-nvm: exited 0.
  ```

  </details>
  
Then, I went to the VSCode web, open the terminal there and look for `.bashrc`
```
abc@49f1d05d2849:~/workspace$ tail ~/.bashrc 
# enable programmable completion features (you don't need to enable
# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
# sources /etc/bash.bashrc).
#if [ -f /etc/bash_completion ] && ! shopt -oq posix; then
#    . /etc/bash_completion
#fi

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
```
Running `nvm` and `node` also works
```
abc@49f1d05d2849:~/workspace$ nvm -v
0.39.1
abc@49f1d05d2849:~/workspace$ node -v
v18.2.0
abc@49f1d05d2849:~/workspace$ npm -v
8.9.0
```

**Review by aptalca — APPROVED** [review]
Great job, works in both code-server and openvscode-server

**Comment by aptalca** [comment]
What happened to the mentions of  `openvscode-server`?

**Comment by fabricionaweb** [comment]
I removed. Gonna add again since it works as well.

**Pull request merged**